src/components/telemetry/moteus/GenericMotorTelemetry.vue

```diff
@@ -79,10 +79,10 @@ function dataCallback(result) {
   let json = JSON.parse(result.json_payload);
 
   for (const key in json) {
-    if (json.hasOwnProperty(key)) {
+    if (key in json) {
```

Comment:
be careful with the `in` keyword. it will match prototype keys, so for parsed json objects, `"toString" in json`, `"valueOf" in json`, `"isPropertyOf" in json`, etc. will all return true. I personally would change it back to `hasOwnProperty` or `Object.hasOwn`.

Reply:
![image](https://github.com/user-attachments/assets/67f40c80-25b0-4322-8f4d-4bb8b69a6774)
@uellenberg I think the eslint and formatted changed this automatically. To note in the future is make sure to the formatter version is the same in tsconfig.json, so it doesn't change it for some reason. I had to add the following in tsconfig.json 
 ```json   
  // Added to allow Object.hasOwn(). We are targeting the latest versions of javascript.
    "target": "ES2023",
    "lib": ["ES2022"]
 ```

 json.hasOwnProperty gives eslint so I think that is why it was automatically changed, so we will be using Object.hasOwn:
`  82:14  error    Do not access Object.prototype method 'hasOwnProperty' from target object                     no-prototype-builtins`

Reply:
Tbh though, hopefully nobody uses prototypes ever... But changed it to keep it clean

Reply:
I did some research into that ESLint error, and it seems that it is valid in that some objects may not have the member function `hasOwnProperty`, so you may want to switch to `Object.hasOwn` 

see: https://stackoverflow.com/questions/39282873/object-hasownproperty-yields-the-eslint-no-prototype-builtins-error-how-to

Reply:
> I did some research into that ESLint error, and it seems that it is valid in that some objects may not have the member function `hasOwnProperty`, so you may want to switch to `Object.hasOwn`
> 
> see: https://stackoverflow.com/questions/39282873/object-hasownproperty-yields-the-eslint-no-prototype-builtins-error-how-to

The problem with hasOwnProperty (at least, from what I remember) is that it's susceptible to prototype pollution, as it exists on the element itself (rather than a global like Object). I don't think it's a huge issue, but if we're worrying about prototype pollution anyway, then we should worry about this as well.

I'm not sure that this is an actual issue though, as (from what I remember) a `for ... in` loop only goes through the enumerable properties of an object - of which prototypes are not (perhaps by default). We should be able to eliminate the if statement entirely, unless there's a specific reason for it (in which case, it should be in a comment).

Reply:
I didn't even see it was in a for in loop, you're probably right that this if check isn't required at all. The mdn docs have a [helpful table[(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties#querying_object_properties) that implies for in is actually more restrictive than `hasOwn` or `hasOwnProperty` so I'm leaning towards removing the loop altogether and just making a note of this in later code.